### PR TITLE
add --custom_channel flag for conda build using custom channel

### PR DESCRIPTION
**Pull request by junpuf**
#4601

### CODE REVIEW

File: tools/pkg-helpers/pytorch_pkg_helpers/__main__.py [hunk 1]
```diff
@@ -33,11 +33,17 @@ def parse_args() -> argparse.Namespace:
...
+        help="Channel to look in, either pytorch-nightly or pytorch-test",
         choices=["nightly", "test"],
         type=str,
         default=os.getenv("CHANNEL", "nightly"),
     )
+    parser.add_argument(
+        "--custom-channel",
+        help="Custom channel to look in, ignore --channel argument when this is set",
+        type=str,
+        default=os.getenv("CUSTOM_CHANNEL", None),
```
Comment (malfet) [inline tools/pkg-helpers/pytorch_pkg_helpers/__main__.py:45]: Do you expect it to be passed via environment variable? Or via command line option

**Review by malfet — CHANGES_REQUESTED** [review]
Please add a check that both `--channel` and `--custom-channel` are not defined at the same time.  (And error out if `custom-channel` is define when this script is invoked for wheel building)

Please get rid of passing default `custom-channel` option via environment variable, unless one plans to use it. 

Is it possible to create some sort fo a unit test for those kinds of changes?

Optional: I know it makes change much more involved, but it would be nice to rename `--channel` to `--pytorch-channel`

File: tools/pkg-helpers/pytorch_pkg_helpers/__main__.py (hunk 1 above)
Comment (junpuf) [reply tools/pkg-helpers/pytorch_pkg_helpers/__main__.py:45]: Hi @malfet thanks for taking a look. We would probably use this option by setting environment variables. I've added both the command line option and environment variable so the code looks consistent with other argument definitions. Let me know if you think we should get rid of the new option and just handle the new environment variable.

Comment (junpuf) [reply tools/pkg-helpers/pytorch_pkg_helpers/__main__.py]: closed this PR for now as we are investigating an issue where `conda search 'pytorch[channel={name}]'` doesn't seems to be working for local file system channel like `file://<path>`

**Pull request closed without merging**